src/prefect/agent.py

```diff
@@ -262,12 +262,6 @@ async def start(self):
         await self.client.__aenter__()
         await self.task_group.__aenter__()
 
-        # Convert the passed default infrastructure to an id
-        if self.default_infrastructure and not self.default_infrastructure_document_id:
-            self.default_infrastructure_document_id = (
-                await self.default_infrastructure._save(is_anonymous=True)
-            )
-
```

Comment:
With this change we'll fail at https://github.com/PrefectHQ/prefect/blob/33350dfb901a994000739bdf5f13338a0c3c1c55/src/prefect/agent.py#L161-L165

Reply:
@madkinsz updated 👍 good catch